AutomatedTesting/Gem/PythonTests/scripting/NodeCategory_ExpandOnClick.py

```diff
@@ -45,79 +55,90 @@ def NodeCategory_ExpandOnClick():
+        helper.wait_for_condition(lambda: self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT) is not None, WAIT_TIME_3)
+        node_palette_widget = self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT)
+        node_palette_node_tree = node_palette_widget.findChild(QtWidgets.QTreeView, TREE_VIEW_QT)
+        node_palette_math_category = pyside_utils.find_child_by_pattern(node_palette_node_tree, NODE_CATEGORY_MATH)
 
-    # 4) Left-Click on a node category arrow to expand it
-    left_click_arrow(nodeTree, ai_index)
+        # 3) Ensure all categories are collapsed for a clean state
+        node_palette_node_tree.collapseAll()
 
-    # 5) Verify it expanded
-    Report.result(Tests.click_expand, nodeTree.isExpanded(ai_index))
+        # 4) Left-Click on a node category arrow to expand it
+        self.left_click_expander_button(node_palette_node_tree, node_palette_math_category)
 
-    # 6) Left-Click on a node category arrow to collapse it
-    left_click_arrow(nodeTree, ai_index)
+        # 5) Verify it expanded
+        category_has_expanded = helper.wait_for_condition(lambda: node_palette_node_tree.isExpanded(node_palette_math_category), WAIT_TIME_3)
+        Report.result(Tests.click_expand, category_has_expanded)
 
-    # 7) Verify it collapsed
-    Report.result(Tests.click_collapse, not nodeTree.isExpanded(ai_index))
+        # 6) Left-Click on a node category arrow to collapse it
+        self.left_click_expander_button(node_palette_node_tree, node_palette_math_category)
 
-    # 8) Double-Click on a node category to expand it
-    double_click(nodeTree, ai_index)
+        # 7) Verify it collapsed
+        category_has_expanded = helper.wait_for_condition(lambda: node_palette_node_tree.isExpanded(node_palette_math_category), WAIT_TIME_3)
+        Report.result(Tests.click_collapse, not category_has_expanded)
 
-    # 9) Verify it expanded
-    Report.result(Tests.dClick_expand, nodeTree.isExpanded(ai_index))
+        # 8) Double-Click on a node category to expand it
+        self.double_click_node_category(node_palette_node_tree, node_palette_math_category)
 
-    # 10) Double-Click on a node category to collapse it
-    double_click(nodeTree, ai_index)
+        # 9) Verify it expanded
+        category_has_expanded = helper.wait_for_condition(lambda: node_palette_node_tree.isExpanded(node_palette_math_category), WAIT_TIME_3)
+        Report.result(Tests.dClick_expand, category_has_expanded)
 
-    # 11) Verify it collapsed
-    Report.result(Tests.dClick_collapse, not nodeTree.isExpanded(ai_index))
+        # 10) Double-Click on a node category to collapse it
+        self.double_click_node_category(node_palette_node_tree, node_palette_math_category)
 
+        # 11) Verify it collapsed
+        category_has_expanded = helper.wait_for_condition(lambda: node_palette_node_tree.isExpanded(node_palette_math_category), WAIT_TIME_3)
+        Report.result(Tests.dClick_collapse, not category_has_expanded)
```

Comment:
Combine action and verification under one comment/step and updated test steps in desc.

Reply:
combined into one step and updated comments